integrations/jenkins/main.py

```diff
@@ -0,0 +1,62 @@
+from typing import Any
+from loguru import logger
+from enum import StrEnum
+
+from client import JenkinsClient
+from port_ocean.context.ocean import ocean
+from port_ocean.core.ocean_types import ASYNC_GENERATOR_RESYNC_TYPE
+
+
+class ObjectKind(StrEnum):
+    JOB = "job"
+    BUILD = "build"
+
+    @staticmethod
+    def get_object_kind_for_event(obj_type: str) -> str | None:
+        if obj_type.startswith("item"):
+            return ObjectKind.JOB
+        elif obj_type.startswith("run"):
+            return ObjectKind.BUILD
+        else:
+            return None
+
+
+def init_client() -> JenkinsClient:
+    return JenkinsClient(
+        ocean.integration_config["jenkins_host"],
+        ocean.integration_config["jenkins_user"],
+        ocean.integration_config["jenkins_token"],
+    )
+
+
+@ocean.on_resync(ObjectKind.JOB)
+async def on_resync_jobs(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
+    jenkins_client = init_client()
+
+    async for jobs in jenkins_client.get_jobs():
+        logger.info(f"Received batch with {len(jobs)} jobs")
+        yield jobs
```

Comment:
you still return the build information from inside the job
please remove it from the object itself

Reply:
is there a particular property you are referrring to? There is no build information in the jobs now. I only query the builds on the appropriate kind

```python
builds_query = (
            ",builds[id,number,url,result,duration,timestamp,displayName,fullDisplayName,previousBuild[id,url]]{0,50}"
            if resource == "builds"
            else ""
        )
 
jobs_query = f"jobs[name,url,displayName,fullName,color,buildable,jobs[name,color,fullName,displayName,url]{builds_query}]"

return {"tree": jobs_query + jobs_pagination}
```